src/components/UserForm.js

```diff
@@ -111,29 +98,30 @@ const AddUserForm = (props) => {
 						<label>Firstname</label>
 						<input
 							type="text"
-							value={fname}
-							onChange={fnameChangeHandler}
+							name="fname"
+							value={formData.fname}
```

Comment:
It is never a valid way to add values. what if formData object do not have fname key? or fname is null? it will set input field undefined.

Better destructure these keys from formData obj, like

const {fname = '', lname=''} = formData and use fname instead of formData.fname